src/main/java/teammates/ui/webapi/action/PostStudentProfilePictureAction.java

```diff
@@ -0,0 +1,98 @@
+package teammates.ui.webapi.action;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.servlet.http.Part;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.exception.EntityNotFoundException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.common.util.GoogleCloudStorageHelper;
+import teammates.ui.webapi.output.ApiOutput;
+
+/**
+ * Action: saves the file information of the profile picture
+ *         that was just uploaded.
+ */
+public class PostStudentProfilePictureAction extends Action {
+    /*
+     * This class is not tested in ActionTests as it is difficult to
+     * reproduce the upload action.
+     * TODO: To cover it in UiTests.
+     */
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isStudent) {
+            throw new UnauthorizedAccessException("Student privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() throws EntityNotFoundException {
+        try {
+            Part image = extractProfilePicture();
+            byte[] imageData = new byte[(int) image.getSize()];
```

Comment:
Potential NPE here if `image` is null.

Reply:
I get a `AvoidCatchingNPE` lint rule if I try to catch NPE. Anyway, in the next line when the image is being read, an `IOException` would be caught if the image is invalid.

Reply:
Catching NPE is of course discouraged. What you should do is:
```java
if (image == null) {
   ...
}
```